### ISSUE TITLE
Problem using output from MSFragger in percolator

### ISSUE THREAD

**Issue by repliko** [percolator/percolator#214]
Hi!

I am using MSFragger to search for PTMs in MS data from a tryptic digest of a purified, recombinant human protein. Now I would like to try using percolator with the output .pepxml file from MSFragger. That is where I run into problems.

MSFragger runs successfully, using an .mzML data file and a human uniprot database with targets and reverse decoys. I then convert the output .pepxml file to .pin using psm-convert with:
> crux psm-convert --input-format pepxml ./data.pepXML pin

Finally I run percolator with:
>percolator -P rev_ ./crux-output/psm-convert.pin

I then get the error message: 
"Hyperparameters: selectionFdr=0.01, Cpos=0, Cneg=0, maxNiter=10
Reading tab-delimited input from datafile ./crux-output/psm-convert.pin
Features:
PepLen Charge1 Charge2 Charge3 Charge4 Charge5 Charge6 Charge7 Charge8 enzN enzC enzInt lnNumSP dM absdM 
Found 5129 PSMs
Concatenated search input detected, skipping both target-decoy competition and mix-max.
Train/test set contains 5129 positives and 0 negatives, size ratio=inf and pi0=1
Exception caught: Error: no decoy PSMs were provided."

I guess the problem is that percolator can not find the decoy results in the output from MSFragger, but I am not sure how to solve it. I am grateful for any input!

Thanks!

Best regards,
Emil

**Comment by MatthewThe** [comment]
This is probably an issue of MSFragger or Crux rather than percolator, but Can you verify that there are decoy hits in your pepXML file?

**Comment by repliko** [comment]
Thanks for such a quick reply!

Yes, I can find decoy hits. For example:
<search_result>
<search_hit peptide="STAIFVDLQQSDGHVTFSQVDISPGIFLYAAR" massdiff="-414.3970" calc_neutral_pep_mass="3641.6836" peptide_next_aa="T" num_missed_cleavages="0" num_tol_term="2" num_tot_proteins="1" tot_num_ions="124" hit_rank="1" num_matched_ions="6" protein="rev_sp|Q9UL01|DSE_HUMAN Dermatan-sulfate epimerase OS=Homo sapiens OX=9606 GN=DSE PE=1 SV=1" peptide_prev_aa="K" is_rejected="0">
<modification_info>
<mod_aminoacid_mass mass="181.0140" position="16"/>
<mod_aminoacid_mass mass="243.0297" position="29"/>
</modification_info>
<search_score name="hyperscore" value="10.047"/>
<search_score name="nextscore" value="10.047"/>
<search_score name="expect" value="3.365e+00"/>
</search_hit>
</search_result>

**Comment by MatthewThe** [comment]
Okay, that looks alright. Can you find PSMs in the .pin file that have a `-1` in the `label` column, if not then `crux psm-convert` probably does not recognize this as a decoy.

I realize the confusion regarding the `-P` flag in percolator: this flag is not used to recognize decoy PSMs, but only to recognize decoy proteins when protein inference is turned on.

**Comment by repliko** [comment]
No, you're right, the .pin file only contains target PSMs...

Is there any other way I can do this? I would really like to try percolator, which was recommended to me, instead of Peptide/ProteinProphet.

**Comment by MatthewThe** [comment]
If you just set the label of the PSMs with only decoy proteins to `-1` it should work.

**Comment by thabangh** [comment]
Matthew,
Do you know where the -1's are coming from in this workflow? Is this
something that should be corrected in Crux?
Bill

**Comment by repliko** [comment]
@MatthewThe 
Thanks! I just changed all labels of the "rev_" peptides in the .pin file to "-1" and now it works.

**Comment by MatthewThe** [comment]
@repliko , great that it works!

@thabangh , the -1's should indeed be added by `crux psm-convert` in this workflow.

**Comment by thabangh** [comment]
Repliko, can you send me an example output file so that I can try to
replicate this behavior for psm-convert?

Thanks.
Bill

**Comment by repliko** [comment]
@thabangh I have attached the input and output file from psm-convert, unmodified. I then manually changed the label for all "rev_" peptides to "-1"  in the output file, before running percolator. The command I used was:
>crux psm-convert --input-format pepxml ./LH_S1803_128.pepXML pin

[input_output.zip](https://github.com/percolator/percolator/files/2071046/input_output.zip)
